### PR TITLE
Update dependencies for site documentation

### PR DESCRIPTION
**Pull request by bondolo**
Motivation:
The documentation produced for the servicetalk.io website uses plugins
and tools which have been updated. Where practical recent versions of
these dependencies should be used.

Modifications:
Plugins and tool dependencies used for generating and verifying the
servicetalk.io site are updated to current or more recent versions.

Result:
Updated versions are used for generating documentation

### CODE REVIEW

**Comment by bondolo** [comment]
I was unable to update Antora to the latest version, 2.3.4, as the local site generated contains broken links. There appears to have been a change in how the site base URL is calculated beginning in 2.2.0.

**Comment by bondolo** [comment]
I found the Antora 2.2.0 release notes which helped me figure out what needed to be changed so I was able to update to latest Antora, 2.3.4.

I compared the generated output of the current 2.0.0 version and the 2.3.4 and there is no significant difference obvious in the HTML, it has surprisingly few differences.

File: docs/generation/gradle/validateSite.gradle [hunk 1]
```diff
@@ -74,7 +73,8 @@ for (location in ["local", "remote"]) {
 
     classpath = configurations.nuValidator
     main = "nu.validator.client.SimpleCommandLineValidator"
-    args "--skip-non-html" // --also-check-css doesn't work (still checks css as html), so limit to html files
+    args "--also-check-css"
```
Comment (tkountis) [inline docs/generation/gradle/validateSite.gradle:76]: 🚀

**Comment by tkountis** [comment]
> I was unable to update Antora to the latest version, 2.3.4, as the local site generated contains broken links. There appears to have been a change in how the site base URL is calculated beginning in 2.2.0.

Should we open a GH issue for that?

**Review by tkountis — APPROVED** [review]
Checkstyle issue, then LGTM

File: docs/generation/gradle/validateSite.gradle (hunk 1 above)
Comment (bondolo) [reply docs/generation/gradle/validateSite.gradle:76]: Just one of the many benefits of updating!

**Comment by idelpivnitskiy** [comment]
Looks great! Thanks for taking care of it ❤️

**Pull request merged**